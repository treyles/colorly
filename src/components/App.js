import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Upload from './Upload';
import Library from './Library';

const testData = [
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2Fa8140b0e-748a-4a8d-a9fe-91324f4174c5?alt=media&token=5ca4aef7-31cd-4389-b5f5-45da4292f057',
    palette: {
      color1: '#37BEB7'
    },
    title: 'Single Color',
    id: 'a8140b0e-748a-4a8d-a9fe-91324f4174c5'
  },
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2F2ec38480-3daf-4c9e-8a68-f2d9de46f6cb?alt=media&token=a8e97814-1650-48d7-9860-690fc936ecf5',
    palette: {
      color1: '#126794',
      color2: '#F8AC8E',
      color3: '#EF6868',
      color4: '#F2C61B',
      color5: '#DDDDDD'
    },
    title: 'Abstract Drawing',
    id: '2ec38480-3daf-4c9e-8a68-f2d9de46f6cb'
  },
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2F0917631b-26f3-46a3-804c-f5ecc1355e39?alt=media&token=d0e2052a-d3f8-4fbc-b430-b7ae259ed9f3',
    palette: {
      color1: '#49124B',
      color2: '#AE53A0',
      color3: '#F5966A',
      color4: '#F04267'
    },
    title: 'Uber Illustration Style',
    id: '0917631b-26f3-46a3-804c-f5ecc1355e39'
  }
];

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      library: testData
    };

    this.addPaletteToLibrary = this.addPaletteToLibrary.bind(this);
  }

  addPaletteToLibrary(obj) {
    const { library } = this.state;
    this.setState({
      library: [obj].concat(library)
    });
  }

  render() {
    const { library } = this.state;

    return (
      <BrowserRouter>
        <Switch>
          <Route
            exact
            path="/"
            render={() => <Library library={library} />}
          />
          <Route
            path="/upload"
            render={({ history }) => (
              <Upload
                history={history}
                addPaletteToLibrary={this.addPaletteToLibrary}
              />
            )}
          />
        </Switch>
      </BrowserRouter>
    );
  }
}

// <div>
//   <Header />
// <Upload addPaletteToLibrary={this.addPaletteToLibrary} />
// </div>
