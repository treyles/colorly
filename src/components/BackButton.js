import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../utils/Icon';

const BackButton = ({ onClick }) => (
  <button className="back" onClick={onClick}>
    <Icon icon="back" />
    <span>BACK TO PALETTES</span>
  </button>
);

BackButton.defaultProps = {
  onClick: () => {}
};

BackButton.propTypes = {
  onClick: PropTypes.func
};

export default BackButton;
