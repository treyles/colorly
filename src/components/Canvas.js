import React from 'react';
import { storage } from '../utils/base';

export default class Canvas extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mousePosition: {},
      imageLoaded: false,
      previewColor: false,
      currentColor: null,
      winWidth: null,
      winHeight: null
    };

    this.handleColorAndPosition = this.handleColorAndPosition.bind(this);
    this.handleCanvasLeave = this.handleCanvasLeave.bind(this);
    this.handleColorPick = this.handleColorPick.bind(this);
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.ctx = this.canvas.getContext('2d');
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
    this.getImageFile();
  }

  updateWindowDimensions() {
    this.setState({
      winWidth: window.innerWidth,
      winHeight: window.innerHeight
    });
  }

  // TODO: name handleColorPreview ???
  handleColorAndPosition(e) {
    const { imageLoaded } = this.state;

    const rect = this.canvas.getBoundingClientRect();
    const actualX = Math.floor(e.clientX - rect.left);
    const actualY = Math.floor(e.clientY - rect.top);

    const pixelData = this.ctx.getImageData(actualX, actualY, 1, 1);
    const { data } = pixelData;
    const pixelColor = `rgba(${data[0]}, ${data[1]}, ${data[2]}, ${data[3]})`;

    if (imageLoaded) {
      this.setState({
        currentColor: pixelColor,
        mousePosition: { x: actualX, y: actualY },
        previewColor: true
      });
    }
  }

  handleCanvasLeave() {
    this.setState({
      previewColor: false
    });
  }

  handleColorPick() {
    const { currentColor } = this.state;
    this.props.makePalette(currentColor);
  }

  drawCanvasImage(image) {
    const heightPadding = 275;
    const widthPadding = 60;
    const containerWidth = this.state.winWidth - widthPadding;
    const containerHeight = this.state.winHeight - heightPadding;
    let imageWidth = image.width;
    let imageHeight = image.height;

    // if any part of the image (original dimensions) overflow
    // the container, shrink dimensions and maintain aspect ratio
    if (imageWidth > containerWidth || imageHeight > containerHeight) {
      // algorithm modified from 1owk3y: https://bit.ly/2JiYzHz
      const wFits = containerWidth / imageWidth;
      const hFits = containerHeight / imageHeight;
      const minFits = wFits > hFits ? hFits : wFits;

      imageWidth *= minFits;
      imageHeight *= minFits;
    }

    this.canvas.width = imageWidth;
    this.canvas.height = imageHeight;

    // TODO: check if this works
    this.ctx.webkitImageSmoothingEnabled = false;
    this.ctx.drawImage(image, 0, 0, imageWidth, imageHeight);
  }

  getImageFile() {
    const imageFile = storage
      .ref()
      .child(`images/${this.props.imageName}`);

    imageFile.getDownloadURL().then(url => {
      const image = new Image();
      image.crossOrigin = 'anonymous';

      image.onload = () => {
        this.drawCanvasImage(image);
        this.setState({ imageLoaded: true });
      };

      image.src = url;
    });
  }

  render() {
    const { mousePosition, previewColor, currentColor } = this.state;

    const containerStyle = {
      top: `${mousePosition.y}px`,
      left: `${mousePosition.x + 10}px`
    };

    const colorStyle = {
      background: currentColor
    };

    return (
      <div className="canvas-wrapper">
        <canvas
          ref={ref => {
            this.canvas = ref;
          }}
          // width={800}
          // height={550}
          onMouseMove={this.handleColorAndPosition}
          onMouseLeave={this.handleCanvasLeave}
          onClick={this.handleColorPick}
        />
        {previewColor && (
          <div className="preview-container" style={containerStyle}>
            <div className="preview-color" style={colorStyle} />
          </div>
        )}
      </div>
    );
  }
}
