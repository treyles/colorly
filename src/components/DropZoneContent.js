import React from 'react';
import PropTypes from 'prop-types';

export default class DropZoneContent extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.handleImageDrop();
  }

  render() {
    const { imageAlert } = this.props;

    return (
      <div className="drop-zone-content">
        <label htmlFor="file-browse" className={imageAlert && 'alert'}>
          {imageAlert ? (
            <h1>{imageAlert}</h1>
          ) : (
            <div>
              <h1>Drag and Drop Image Here</h1>
              <h3>JPG, GIF, PNG or SVG no larger than 3 MB</h3>
            </div>
          )}
        </label>
        <input
          id="file-browse"
          type="file"
          accept="image/*"
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

DropZoneContent.defaultProps = {
  imageAlert: null
};

DropZoneContent.propTypes = {
  imageAlert: PropTypes.string,
  handleImageDrop: PropTypes.func.isRequired
};
