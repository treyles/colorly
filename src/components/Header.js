/* eslint-disable */
import React from 'react';
import Icon from '../utils/Icon';
import { Link } from 'react-router-dom';
import firebase from 'firebase';
import ClickOutside from '../utils/ClickOutside';
// import Icon from '../utils/Icon';

export default class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profileOpen: false
    };

    this.handleSignOutClick = this.handleSignOutClick.bind(this);
    this.handleDialogToggleClick = this.handleDialogToggleClick.bind(this);
    this.handleAddPaletteClick = this.handleAddPaletteClick.bind(this);
    this.closeDialog = this.closeDialog.bind(this);
  }

  handleSignOutClick() {
    firebase
      .auth()
      .signOut()
      .then(() => {
        console.log('signed out');
        localStorage.setItem('authenticated', false);
      });
  }

  handleDialogToggleClick() {
    this.setState({
      profileOpen: !this.state.profileOpen
    });
  }

  handleAddPaletteClick() {
    if (this.props.isNewUser) {
      this.props.closeNewUserDialog();
    }

    this.props.history.push('/upload');
  }

  closeDialog() {
    this.setState({
      profileOpen: false
    });
  }

  render() {
    const { profileOpen } = this.state;
    const { currentUser, library, animateHeader } = this.props;

    const profileDialog = (
      <div className="profile-dialog">
        <div className="palette-count">
          <span className="count">{library.length}</span>
          {library.length !== 1 ? 'PALETTES' : 'PALETTE'}
        </div>
        <button className="sign-out" onClick={this.handleSignOutClick}>
          SIGN OUT
        </button>
      </div>
    );

    return (
      <header>
        <div
          className="header-top"
          style={{ backgroundColor: `${animateHeader ? '#fff' : ''}` }}
        >
          <div className="profile">
            <div className="profile-image">
              {currentUser.photoURL && <img src={currentUser.photoURL} />}
            </div>
            <span onClick={this.handleDialogToggleClick}>
              {currentUser.displayName}
            </span>
            <span onClick={this.handleDialogToggleClick}>
              <Icon icon="down" />
            </span>
            <ClickOutside
              elementIsOpen={profileOpen}
              onRequestClose={this.closeDialog}
            >
              {profileDialog}
            </ClickOutside>
          </div>
          {/* <Link to="/upload"> */}
          <button className="add" onClick={this.handleAddPaletteClick}>
            {/* <span>
                <Icon icon="add" />
              </span> */}
            ADD PALETTE
          </button>
          {/* </Link> */}
        </div>
        <div
          className="header-logo"
          style={{
            opacity: `${animateHeader ? 0 : ''}`,
            visibility: `${animateHeader ? 'hidden' : 'visible'}`
          }}
        >
          <h2>Colorly</h2>
          <span>
            <Icon icon="logo" />
          </span>
        </div>
      </header>
    );
  }
}

// style={{ background: `${animateHeader ? '#fff' : ''}` }}
// style={{ opacity: `${animateHeader ? 0 : ''}` }}
