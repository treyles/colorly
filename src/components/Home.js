import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../utils/Icon';
import { googleAuth } from '../utils/base';

export default function Home({ signInUser }) {
  return (
    <div className="home">
      <div className="hero-text-container">
        <div className="content">
          <h1>Colorly</h1>
          <span className="logo">
            <Icon icon="logo" />
          </span>
          <h3 className="pitch">
            A simple color extraction tool. Save custom palettes to your
            collection along with the source image for future reference!
          </h3>
          <div className="buttons">
            <button className="twitter-login">Connect with Twitter</button>
            <button
              className="google-login"
              onClick={() => signInUser(googleAuth)}
            >
              Connect with Google
            </button>
          </div>
        </div>
      </div>
      <div className="hero-image" />
    </div>
  );
}

Home.propTypes = {
  signInUser: PropTypes.func.isRequired
};
