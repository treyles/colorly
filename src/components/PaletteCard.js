// native way to set space between grid?
// change color of text light or dark based on bg color
// finish up card!

import React from 'react';

export default class PaletteCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      foo: true
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({
      foo: !this.state.foo
    });
  }

  render() {
    const { title, palette } = this.props.data;

    return (
      <div className="palette-card">
        <div className="palette-colors">
          {Object.keys(palette).map(el => (
            <div
              key={palette[el]}
              className="swatch"
              style={{ background: `${palette[el]}` }}
            />
          ))}
        </div>
        <div className="palette-footer">{title}</div>
      </div>
    );
  }
}
