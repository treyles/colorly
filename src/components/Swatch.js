import React from 'react';
import PropTypes from 'prop-types';
import { rgbToHex, parseRgb } from '../utils/helpers';

export default class Swatch extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      textColor: ''
    };

    this.handleClick = this.handleClick.bind(this);
    this.setInputRef = this.setInputRef.bind(this);
  }

  componentDidMount() {
    this.setTextContrast();
  }

  // check for sufficient contrast
  // between text and background color
  setTextContrast() {
    const rgb = parseRgb(this.props.color);
    const brightnessThreshold = 220;

    // algorithm to determine color brightness
    // https://www.w3.org/TR/AERT/#color-contrast
    const brightness = Math.round(
      (Number(rgb[0]) * 299 +
        Number(rgb[1]) * 587 +
        Number(rgb[2]) * 114) /
        1000
    );

    if (brightness > brightnessThreshold) {
      this.setState({ textColor: '#666' });
    }
  }

  setInputRef(ref) {
    this.hexInput = ref;
  }

  handleClick() {
    this.hexInput.select();
    document.execCommand('copy');
    this.props.activateCopyAlert();
  }

  render() {
    const hex = rgbToHex(this.props.color);

    return (
      <div
        className="swatch"
        style={{ background: `${hex}` }}
        onClick={this.handleClick}
        onKeyDown={this.handleClick}
        role="button"
        tabIndex={0}
      >
        <div className="pop-up" style={{ background: `${hex}` }} />
        <input
          ref={this.setInputRef}
          style={{ color: `${this.state.textColor}` }}
          value={`${hex}`}
          readOnly
        />
      </div>
    );
  }
}

Swatch.propTypes = {
  activateCopyAlert: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired
};
