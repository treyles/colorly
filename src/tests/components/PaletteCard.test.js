import React from 'react';
import { shallow } from 'enzyme';
import { PaletteCard } from '../../components/PaletteCard';
import Swatch from '../../components/Swatch';

describe('PaletteCard', () => {
  let component;
  const props = {
    uid: 'uid',
    data: {
      url: '',
      palette: {
        color1: 'color1',
        color2: 'color2',
        color3: 'color3'
      },
      title: ''
    },
    setImageSource: jest.fn()
  };

  beforeEach(() => {
    component = shallow(<PaletteCard {...props} />);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('toggles optionsOpen state when options button is clicked', () => {
    component.find('.options').simulate('click');
    expect(component.state().optionsOpen).toEqual(true);
  });

  it('renders copied message when copyAlert state is truthy', () => {
    component.instance().activateCopyAlert();
    expect(component.find('.copy-alert').exists()).toEqual(true);
  });

  it('resets copyAlert after 1.5 seconds', () => {
    jest.useFakeTimers();
    component.instance().activateCopyAlert();
    expect(component.state().copyAlert).toEqual(true);
    setTimeout(() => {
      expect(component.state().copyAlert).toEqual(false);
    }, 1500);
    jest.runAllTimers();
  });

  it('renders correct amount of <Swatch /> components based on palette ', () => {
    expect(component.find(Swatch).length).toEqual(3);
  });

  it('calls setImageSource when VIEW button is clicked', () => {
    component.find('.image-view-btn').simulate('click');
    expect(props.setImageSource).toHaveBeenCalled();
  });

  it('calls deleteCardFromLibrary when DELETE button is clicked', () => {
    const spy = jest
      .spyOn(PaletteCard.prototype, 'deleteCardFromLibrary')
      .mockImplementation(jest.fn());
    const wrapper = shallow(<PaletteCard {...props} />);
    wrapper.find('.delete-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
