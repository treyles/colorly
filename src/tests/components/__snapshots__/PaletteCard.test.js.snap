// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaletteCard
    data={
      Object {
        "palette": Object {
          "color1": "color1",
          "color2": "color2",
          "color3": "color3",
        },
        "title": "",
        "url": "",
      }
    }
    deleteCardFromLibrary={[MockFunction]}
    setImageSource={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="palette-swatches"
        >
          <Swatch
            activateCopyAlert={[Function]}
            color="color1"
          />
          <Swatch
            activateCopyAlert={[Function]}
            color="color2"
          />
          <Swatch
            activateCopyAlert={[Function]}
            color="color3"
          />
        </div>,
        <div
          className="palette-footer"
        >
          
          <button
            className="options"
            onClick={[Function]}
          >
            <Icon
              icon="options"
            />
          </button>
          <ClickOutside
            elementIsOpen={false}
            onRequestClose={[Function]}
          >
            <div
              className="options-dialog"
            >
              <button
                className="image-view-btn"
                onClick={[Function]}
              >
                VIEW IMAGE
              </button>
              <button
                className="delete-btn"
                onClick={[Function]}
              >
                DELETE
              </button>
            </div>
          </ClickOutside>
        </div>,
      ],
      "className": "palette-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Swatch
              activateCopyAlert={[Function]}
              color="color1"
            />,
            <Swatch
              activateCopyAlert={[Function]}
              color="color2"
            />,
            <Swatch
              activateCopyAlert={[Function]}
              color="color3"
            />,
          ],
          "className": "palette-swatches",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "activateCopyAlert": [Function],
              "color": "color1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "activateCopyAlert": [Function],
              "color": "color2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "activateCopyAlert": [Function],
              "color": "color3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <button
              className="options"
              onClick={[Function]}
            >
              <Icon
                icon="options"
              />
            </button>,
            <ClickOutside
              elementIsOpen={false}
              onRequestClose={[Function]}
            >
              <div
                className="options-dialog"
              >
                <button
                  className="image-view-btn"
                  onClick={[Function]}
                >
                  VIEW IMAGE
                </button>
                <button
                  className="delete-btn"
                  onClick={[Function]}
                >
                  DELETE
                </button>
              </div>
            </ClickOutside>,
            false,
          ],
          "className": "palette-footer",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                icon="options"
              />,
              "className": "options",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "options",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="options-dialog"
              >
                <button
                  className="image-view-btn"
                  onClick={[Function]}
                >
                  VIEW IMAGE
                </button>
                <button
                  className="delete-btn"
                  onClick={[Function]}
                >
                  DELETE
                </button>
              </div>,
              "elementIsOpen": false,
              "onRequestClose": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="image-view-btn"
                    onClick={[Function]}
                  >
                    VIEW IMAGE
                  </button>,
                  <button
                    className="delete-btn"
                    onClick={[Function]}
                  >
                    DELETE
                  </button>,
                ],
                "className": "options-dialog",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "VIEW IMAGE",
                    "className": "image-view-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "VIEW IMAGE",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DELETE",
                    "className": "delete-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "DELETE",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="palette-swatches"
          >
            <Swatch
              activateCopyAlert={[Function]}
              color="color1"
            />
            <Swatch
              activateCopyAlert={[Function]}
              color="color2"
            />
            <Swatch
              activateCopyAlert={[Function]}
              color="color3"
            />
          </div>,
          <div
            className="palette-footer"
          >
            
            <button
              className="options"
              onClick={[Function]}
            >
              <Icon
                icon="options"
              />
            </button>
            <ClickOutside
              elementIsOpen={false}
              onRequestClose={[Function]}
            >
              <div
                className="options-dialog"
              >
                <button
                  className="image-view-btn"
                  onClick={[Function]}
                >
                  VIEW IMAGE
                </button>
                <button
                  className="delete-btn"
                  onClick={[Function]}
                >
                  DELETE
                </button>
              </div>
            </ClickOutside>
          </div>,
        ],
        "className": "palette-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Swatch
                activateCopyAlert={[Function]}
                color="color1"
              />,
              <Swatch
                activateCopyAlert={[Function]}
                color="color2"
              />,
              <Swatch
                activateCopyAlert={[Function]}
                color="color3"
              />,
            ],
            "className": "palette-swatches",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "activateCopyAlert": [Function],
                "color": "color1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "activateCopyAlert": [Function],
                "color": "color2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "activateCopyAlert": [Function],
                "color": "color3",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <button
                className="options"
                onClick={[Function]}
              >
                <Icon
                  icon="options"
                />
              </button>,
              <ClickOutside
                elementIsOpen={false}
                onRequestClose={[Function]}
              >
                <div
                  className="options-dialog"
                >
                  <button
                    className="image-view-btn"
                    onClick={[Function]}
                  >
                    VIEW IMAGE
                  </button>
                  <button
                    className="delete-btn"
                    onClick={[Function]}
                  >
                    DELETE
                  </button>
                </div>
              </ClickOutside>,
              false,
            ],
            "className": "palette-footer",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  icon="options"
                />,
                "className": "options",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "icon": "options",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="options-dialog"
                >
                  <button
                    className="image-view-btn"
                    onClick={[Function]}
                  >
                    VIEW IMAGE
                  </button>
                  <button
                    className="delete-btn"
                    onClick={[Function]}
                  >
                    DELETE
                  </button>
                </div>,
                "elementIsOpen": false,
                "onRequestClose": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="image-view-btn"
                      onClick={[Function]}
                    >
                      VIEW IMAGE
                    </button>,
                    <button
                      className="delete-btn"
                      onClick={[Function]}
                    >
                      DELETE
                    </button>,
                  ],
                  "className": "options-dialog",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "VIEW IMAGE",
                      "className": "image-view-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "VIEW IMAGE",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "DELETE",
                      "className": "delete-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "DELETE",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
